server/src/main/java/org/elasticsearch/action/search/SearchShardsGroup.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.search;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.shard.ShardId;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A group of target shards of a response of the search_shards API
```

Comment:
I tend to get confused about the group terminology here: is it to indicate that we track the shard_id, and the nodes where all of its copies are allocated to (aka replication group)?

Reply:
I revised this javadoc in https://github.com/elastic/elasticsearch/pull/94534/commits/3f671cdfa64ccb933b088ec8d88cb056a473b44a